central/complianceoperator/v2/report/manager/results/results_aggregator.go

```diff
@@ -65,36 +67,96 @@ type aggregateResultsFn func(context.Context, string, *[]*report.ResultRow, *che
 func (g *Aggregator) GetReportData(req *report.Request) *report.Results {
 	resultsCSV := make(map[string][]*report.ResultRow)
 	reportResults := &report.Results{}
+	reportResults.ClustersData = make(map[string]*report.ClusterData)
 	for _, clusterID := range req.ClusterIDs {
-		clusterResults, clusterStatus, err := g.getReportDataForCluster(req.Ctx, req.ScanConfigID, clusterID, req.FailedClusters)
+		clusterData, err := g.getClusterData(req.Ctx, req.ScanConfigID, clusterID, req.ClusterData[clusterID])
+		if err != nil {
+			log.Errorf("Data unable to enhance cluster data for cluster %q: %v", clusterID, err)
+			continue
+		}
+		clusterResults, clusterStatus, err := g.getReportDataForCluster(req.Ctx, req.ScanConfigID, clusterID, clusterData)
 		if err != nil {
 			log.Errorf("Data not found for cluster %s", clusterID)
 			continue
 		}
+
 		resultsCSV[clusterID] = clusterResults
 		reportResults.TotalPass += clusterStatus.totalPass
 		reportResults.TotalFail += clusterStatus.totalFail
 		reportResults.TotalMixed += clusterStatus.totalMixed
+		reportResults.ClustersData[clusterID] = clusterData
 	}
 	reportResults.Clusters = len(req.ClusterIDs)
 	reportResults.Profiles = req.Profiles
 	reportResults.ResultCSVs = resultsCSV
 	return reportResults
 }
 
-func (g *Aggregator) getReportDataForCluster(ctx context.Context, scanConfigID, clusterID string, failedClusters map[string]*storage.ComplianceOperatorReportSnapshotV2_FailedCluster) ([]*report.ResultRow, *checkStatus, error) {
+func (g *Aggregator) getClusterData(ctx context.Context, scanConfigID, clusterID string, clusterData *report.ClusterData) (*report.ClusterData, error) {
+	if clusterData == nil {
+		return nil, errors.Errorf("cluster data for cluster %q is nil", clusterID)
+	}
+
+	scanNamesToProfileNames, err := g.scanDS.GetProfilesScanNamesByScanConfigAndCluster(ctx, scanConfigID, clusterID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to retrieve profiles associated with the ScanConfiguration %q in the cluster %q", scanConfigID, clusterID)
+	}
+	var profileNames []string
+	var scanNames []string
+	for scanName, profileName := range scanNamesToProfileNames {
+		profileNames = append(profileNames, profileName)
+		scanNames = append(scanNames, scanName)
+	}
+	clusterData.Profiles = profileNames
+	clusterData.Scans = scanNames
```

Comment:
Would that work here?
```suggestion
	clusterData.Profiles = maps.Keys(scanNamesToProfileNames)
	clusterData.Scans = maps.Values(scanNames)
```
We could get rid of the entire for loop and copying if possible.